RecoBTag/ONNXRuntime/plugins/BoostedJetONNXJetTagsProducer.cc

```diff
@@ -44,69 +45,123 @@ class DeepBoostedJetONNXJetTagsProducer : public edm::stream::EDProducer<edm::Gl
-      double replace_inf_value = var_pset.getParameter<double>("replace_inf_value");
-      double lower_bound = var_pset.getParameter<double>("lower_bound");
-      double upper_bound = var_pset.getParameter<double>("upper_bound");
-      prep_params.var_info_map[var_name] =
-          PreprocessParams::VarInfo(median, norm_factor, replace_inf_value, lower_bound, upper_bound);
+  auto json_path = iConfig.getParameter<std::string>("preprocess_json");
+  if (!json_path.empty()) {
+    // use preprocessing json file if available
+    std::ifstream ifs(edm::FileInPath(json_path).fullPath());
+    nlohmann::json j = nlohmann::json::parse(ifs);
+    j.at("input_names").get_to(input_names_);
+    for (const auto &group_name : input_names_) {
+      const auto &group_pset = j.at(group_name);
+      auto &prep_params = prep_info_map_[group_name];
+      group_pset.at("var_names").get_to(prep_params.var_names);
+      if (group_pset.contains("var_length")) {
+        prep_params.min_length = group_pset.at("var_length");
+        prep_params.max_length = prep_params.min_length;
+      } else {
+        prep_params.min_length = group_pset.at("min_length");
+        prep_params.max_length = group_pset.at("max_length");
+        input_shapes_.push_back({1, (int64_t)prep_params.var_names.size(), -1});
+      }
+      const auto &var_info_pset = group_pset.at("var_infos");
+      for (const auto &var_name : prep_params.var_names) {
+        const auto &var_pset = var_info_pset.at(var_name);
+        double median = var_pset.at("median");
+        double norm_factor = var_pset.at("norm_factor");
+        double replace_inf_value =
+            var_pset.contains("replace_inf_value") ? double(var_pset.at("replace_inf_value")) : 0;
+        double lower_bound = var_pset.contains("lower_bound") ? double(var_pset.at("lower_bound")) : -5;
+        double upper_bound = var_pset.contains("upper_bound") ? double(var_pset.at("upper_bound")) : 5;
+        double pad = var_pset.contains("pad") ? double(var_pset.at("pad")) : 0;
+        prep_params.var_info_map[var_name] =
+            PreprocessParams::VarInfo(median, norm_factor, replace_inf_value, lower_bound, upper_bound, pad);
+      }
+
+      // create data storage with a fixed size vector initilized w/ 0
+      const auto &len = input_sizes_.emplace_back(prep_params.max_length * prep_params.var_names.size());
+      data_.emplace_back(len, 0);
     }
+  } else {
+    // otherwise use the PSet in the python config file
+    const auto &prep_pset = iConfig.getParameterSet("preprocessParams");
+    input_names_ = prep_pset.getParameter<std::vector<std::string>>("input_names");
+    for (const auto &group_name : input_names_) {
+      const auto &group_pset = prep_pset.getParameterSet(group_name);
+      auto &prep_params = prep_info_map_[group_name];
+      prep_params.var_names = group_pset.getParameter<std::vector<std::string>>("var_names");
+      if (group_pset.existsAs<unsigned>("var_length")) {
```

Comment:
::exists method is rather an exception left for cases for rather old and complex pre-7X software.
Define a ::fillDescriptions method https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp

the earlier version did not have existsAs
and apparently there are no new parameter files.
Considering that `preprocessParams is deprecated `, I think that this flexible support is not needed and can be removed.